File: MoonEngine-Root/OpenGL/src/main/java/com/greentree/engine/moon/opengl/adapter/FreamBufferCommandBuffer.java
package com.greentree.engine.moon.opengl.adapter;

import static org.lwjgl.opengl.GL11.*;

import java.util.Objects;

import com.greentree.common.graphics.sgl.freambuffer.FreamBuffer;
import com.greentree.engine.moon.render.pipeline.target.buffer.ProxyCommandBuffer;
import com.greentree.engine.moon.render.pipeline.target.buffer.TargetCommandBuffer;


public class FreamBufferCommandBuffer extends ProxyCommandBuffer {
	
	protected final FreamBuffer fb;
	
	public FreamBufferCommandBuffer(FreamBuffer fb, TargetCommandBuffer buffer) {
		super(buffer);
		Objects.requireNonNull(fb);
		this.fb = fb;
	}
	
	@Override
	public void close() {
		fb.bind();
		glViewport(0, 0, fb.getWidth(), fb.getHeight());
		super.close();
		fb.unbind();
	}
	
}

File: MoonEngine-Root/OpenGL/src/main/java/com/greentree/engine/moon/opengl/adapter/FreamBufferCommandBuffer.kt
package com.greentree.engine.moon.opengl.adapter

import com.greentree.common.graphics.sgl.freambuffer.FreamBuffer
import com.greentree.commons.graphics.smart.target.RenderCommandBuffer
import org.lwjgl.opengl.GL11

class FreamBufferCommandBuffer(val fb: FreamBuffer, val origin: RenderCommandBuffer) : RenderCommandBuffer by origin {

	override fun execute() {
		fb.bind()
		GL11.glViewport(0, 0, fb.width, fb.height)
		origin.execute()
		fb.unbind()
	}
}